### ISSUE TITLE
Running "./utils/restore-profiles.sh localhost:8080" does not create all profiles

### ISSUE THREAD

**Issue by jrabek** [facebook/augmented-traffic-control#119]
Followed the instructions on https://github.com/facebook/augmented-traffic-control.
After running "./utils/restore-profiles.sh localhost:8080", the profiles are not all created.  Most of the time no profiles are created.

The only time profiles actually show up in the web interface is when curl gets a response like the following:
```
jrabek:~/projects/atc: (master) $ ./utils/restore-profiles.sh localhost:8080
Adding profile /Users/jrabek/projects/atc/utils/profiles/2G-DevelopingRural.json
Adding profile /Users/jrabek/projects/atc/utils/profiles/2G-DevelopingUrban.json
Adding profile /Users/jrabek/projects/atc/utils/profiles/3G-Average.json
Adding profile /Users/jrabek/projects/atc/utils/profiles/3G-Good.json
Adding profile /Users/jrabek/projects/atc/utils/profiles/Cable.json
Adding profile /Users/jrabek/projects/atc/utils/profiles/DSL.json
{"id":4,"name":"DSL","content":{"down":{"loss":{"percentage":0,"correlation":0},"delay":{"delay":"5","jitter":0,"correlation":0},"rate":"2000","iptables_options":[],"corruption":{"percentage":0,"correlation":0},"reorder":{"percentage":0,"correlation":0,"gap":0}},"up":{"loss":{"percentage":0,"correlation":0},"delay":{"delay":"5","jitter":0,"correlation":0},"rate":"256","iptables_options":[],"corruption":{"percentage":0,"correlation":0},"reorder":{"percentage":0,"correlation":0,"gap":0}}}}Adding profile /Users/jrabek/projects/atc/utils/profiles/Edge-Average.json
Adding profile /Users/jrabek/projects/atc/utils/profiles/Edge-Good.json
Adding profile /Users/jrabek/projects/atc/utils/profiles/Edge-Lossy.json
Adding profile /Users/jrabek/projects/atc/utils/profiles/NoConnectivity.json
```
The only profile that showed up in the UI in this case was the DSL profile.

No errors show up in /var/log/atc/atcui.log in the virtual box instance.  Not sure where the atcd logs show up.

**Comment by chantra** [comment]
@jrabek 
Thanks for reporting this. @zfjagann and I are haven't had a chance to look at it yet at there is a long holiday weekend here. Assigning to @zfjagann to look at it and potentially add more verbose logging upon failure.

**Comment by zfjagann** [comment]
Hmm.... It works for me. I am, however, adding some error checking and it now prints the curl exit code if it fails like this:

    Could not add profile /home/zeal/dev/fb/atc/utils/profiles/NoConnectivity.json (curl exit code 6)

Whereas a success:

    Added profile /home/zeal/dev/fb/atc/utils/profiles/NoConnectivity.json

Please try it again with the updated script and let me know if it's still having issues.

The ATC UI logs should go into both `/var/log/upstart/atcui.log` and `/var/log/atc/atcui.log`. You sould check both of those locations for error messages.

**Comment by zfjagann** [comment]
[Updated script](https://github.com/facebook/augmented-traffic-control/blob/master/utils/restore-profiles.sh)

**Comment by jrabek** [comment]
@zfjagann Here is the output of the updated script:
```
jrabek:~/projects/atc: (master) $ ./utils/restore-profiles.sh localhost:8080
Could not add profile /Users/jrabek/projects/atc/utils/profiles/2G-DevelopingRural.json (curl exit code 7)
Could not add profile /Users/jrabek/projects/atc/utils/profiles/2G-DevelopingUrban.json (curl exit code 7)
Could not add profile /Users/jrabek/projects/atc/utils/profiles/3G-Average.json (curl exit code 7)
Could not add profile /Users/jrabek/projects/atc/utils/profiles/3G-Good.json (curl exit code 7)
Could not add profile /Users/jrabek/projects/atc/utils/profiles/Cable.json (curl exit code 7)
Could not add profile /Users/jrabek/projects/atc/utils/profiles/DSL.json (curl exit code 7)
Added profile /Users/jrabek/projects/atc/utils/profiles/Edge-Average.json
Could not add profile /Users/jrabek/projects/atc/utils/profiles/Edge-Good.json (curl exit code 7)
Could not add profile /Users/jrabek/projects/atc/utils/profiles/Edge-Lossy.json (curl exit code 7)
Could not add profile /Users/jrabek/projects/atc/utils/profiles/NoConnectivity.json (curl exit code 7)
```

**Comment by zfjagann** [comment]
Judging from the [exit code 7](http://curl.haxx.se/libcurl/c/libcurl-errors.html#CURLECOULDNTCONNECT) looks like it can't connect to the port. Can you `curl http://localhost:8080/` manually?

**Comment by jrabek** [comment]
@zfjagann I tailed the logs you mentioned and did not see any new logs while running restore-profiles.sh.
```
sudo tail -f /var/log/upstart/atcui.log
```
```
sudo tail -f /var/log/atc/atcui.log
```

I did see some errors in the atcui.log from earlier:
```
ImportError: No module named atc_api
Traceback (most recent call last):
  File "/usr/local/atc/venv/local/lib/python2.7/site-packages/gunicorn/arbiter.py", line 507, in spawn_worker
    worker.init_process()
  File "/usr/local/atc/venv/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 118, in init_process
    self.wsgi = self.app.wsgi()
  File "/usr/local/atc/venv/local/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/usr/local/atc/venv/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/usr/local/atc/venv/local/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/usr/local/atc/venv/local/lib/python2.7/site-packages/gunicorn/util.py", line 355, in import_app
    __import__(module)
  File "/var/django/atc_ui/wsgi.py", line 14, in <module>
    application = get_wsgi_application()
  File "/usr/local/atc/venv/local/lib/python2.7/site-packages/django/core/wsgi.py", line 14, in get_wsgi_application
    django.setup()
  File "/usr/local/atc/venv/local/lib/python2.7/site-packages/django/__init__.py", line 21, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/atc/venv/local/lib/python2.7/site-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/usr/local/atc/venv/local/lib/python2.7/site-packages/django/apps/config.py", line 87, in create
    module = import_module(entry)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named atc_api
[2015-05-26 16:43:23 +0000] [1251] [INFO] Worker exiting (pid: 1251)
[2015-05-26 23:45:37 +0000] [1610] [INFO] Starting gunicorn 19.3.0
[2015-05-26 23:45:37 +0000] [1610] [INFO] Listening at: http://0.0.0.0:8000 (1610)
[2015-05-26 23:45:37 +0000] [1610] [INFO] Using worker: sync
[2015-05-26 23:45:37 +0000] [1615] [INFO] Booting worker with pid: 1615
[2015-05-26 23:45:37 +0000] [1616] [INFO] Booting worker with pid: 1616
```

**Comment by jrabek** [comment]
@zfjagann So interesting discovery:
This works 100% of the time and loads all profiles:
```
./utils/restore-profiles.sh 127.0.0.1:8080
```
Whereas this only loads the profiles sometimes
```
./utils/restore-profiles.sh localhost:8080
```

curl seems to have a problem resolving localhost for me.  Not sure why.  Using 127.0.0.1 works fine.

You can close this issue.  Maybe change the instructions for restore-profiles.sh to use a numeric address instead of localhost to avoid the lookup?  The strange part is that my web browser never has a problem.  Just curl.  Maybe an Apple weirdness?

```
jrabek:~/projects/atc: (master) $ curl --version
curl 7.37.1 (x86_64-apple-darwin14.0) libcurl/7.37.1 SecureTransport zlib/1.2.5
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smtp smtps telnet tftp
Features: AsynchDNS GSS-Negotiate IPv6 Largefile NTLM NTLM_WB SSL libz
```

**Comment by zfjagann** [comment]
Hmm. That's curious.

My guess is localhost resolves to 127.0.1.1 instead of 127.0.0.1 (I believe this is the default in ubuntu, but I don't know why). Check your `/etc/hosts` file and `nslookup localhost` to be sure.

Glad it got figured out.
